server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -191,26 +275,21 @@ runQueryDB
      , MonadReader E.ExecutionCtx m
      , MonadQueryLog m
      , MonadTrace m
-     , MonadExecuteQuery m
      )
   => RequestId
-  -> (GQLReqUnparsed, GQLReqParsed)
-  -> [QueryRootField UnpreparedValue]
+  -> GQLReqUnparsed
   -> UserInfo
-  -> (Tracing.TraceT (LazyTx QErr) EncJSON, EQ.GeneratedSqlMap)
-  -> m (DiffTime, Telem.QueryType, HTTP.ResponseHeaders, EncJSON)
-  -- ^ Also return 'Mutation' when the operation was a mutation, and the time
-  -- spent in the PG query; for telemetry.
-runQueryDB reqId (query, queryParsed) asts _userInfo (tx, genSql) =  do
+  -> Text
+  -> (Tracing.TraceT (LazyTx QErr) EncJSON, Maybe EQ.GeneratedSqlMap)
```

Comment:
It seems like the `GeneratedSqlMap` here only ever contains a single element, and the `fieldName` is given elsewhere anyway. Could this just accept a chunk of SQL? The map wrapper seems unnecessary.

Reply:
Okay, I've pushed the responsibility of constructing the hashmap down to `runQueryDB` so that it can take a `PreparedSql` argument instead. We can't get rid of the hashmap entirely without further discussion as this is used in the type class `MonadQueryLog` which is also used by other codebases.

Reply:
Do we ever actually call `logQueryLog` with a map containing more than one entry in it? If not, I’d be in favor of just changing the signature—I don’t think the tiny bit of extra downstream effort to update to the new interface is worth the misleading API. (Of course, if we *do* call it with more than one value somewhere, please point me to it!)

Reply:
We wanted to avoid doing too many changes for pro; since there was no direct need to change `logQueryLog` for the purpose of this PR, we chose not to change anything.
However, while @paf31 agreed with the changes as first done in #5865, we're thinking of doing a second PR to do some cleanup, after this one is submitted and further work on dataloader is unblocked. We could definitely include that change as part of the second PR!

Reply:
After discussion: pushed a change to `MonadQueryLog`.